src/Common/AzurePSCmdlet.cs

```diff
@@ -322,6 +322,17 @@ protected virtual void SetupHttpClientPipeline()
             AzureSession.Instance.ClientFactory.AddUserAgent("AzurePowershell", string.Format("v{0}", AzVersion));
             AzureSession.Instance.ClientFactory.AddUserAgent(PSVERSION, string.Format("v{0}", PowerShellVersion));
             AzureSession.Instance.ClientFactory.AddUserAgent(ModuleName, this.ModuleVersion);
+            try {
+                string hostEnv = AzurePSCmdlet.getEnvUserAgent();
+                if (!String.IsNullOrWhiteSpace(hostEnv)) 
+                {
+                    AzureSession.Instance.ClientFactory.AddUserAgent(hostEnv);
+                }
+            } 
+            catch (Exception) 
```

Comment:
If we are try-catch `AddUserAgent()`, does it make sense to wrap all `AddUserAgent()` calls in it?

Reply:
> If we are try-catch `AddUserAgent()`, does it make sense to wrap all `AddUserAgent()` calls in it?

Here is `AzureSession.Instance.ClientFactory.AddUserAgent(hostEnv);`. Below is `ClientFactory.RemoveUserAgent(hostEnv);`. The Client part is `UserAgent += string.Format(" {0}", hostEnv);` I can't merge them together.